components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java

```diff
@@ -298,18 +368,31 @@ public IObjectFactory getObjectFactory() {
         return new org.powermock.modules.testng.PowerMockObjectFactory();
     }
 
-    private void createOAuthClientRequest(OAuthClientRequest.TokenRequestBuilder mockTokenRequestBuilder,
-                                          OAuthClientRequest mockOAuthClientRequest) throws OAuthSystemException {
-        when(OAuthClientRequest.tokenLocation(anyString())).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.setGrantType(GrantType.AUTHORIZATION_CODE)).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.setClientId(anyString())).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.setClientSecret(anyString())).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.setRedirectURI(anyString())).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.setCode(anyString())).thenReturn(mockTokenRequestBuilder);
-        when(mockTokenRequestBuilder.buildBodyMessage()).thenReturn(mockOAuthClientRequest);
+    private void setupTest() throws OAuthProblemException, AuthenticationFailedException, UserStoreException {
+        mockStatic(OAuthAuthzResponse.class);
+        when(OAuthAuthzResponse.oauthCodeAuthzResponse(mockServletRequest)).thenReturn(mockOAuthzResponse);
+        when(mockServletRequest.getParameter("domain")).thenReturn("carbon.super");
+        mockAuthenticationRequestContext(mockAuthenticationContext);
+        when(mockOAuthzResponse.getCode()).thenReturn("200");
+        doReturn(mockOAuthClientResponse).when(openIDConnectAuthenticator).getOauthResponse(any(OAuthClient.class), any(OAuthClientRequest.class));
+        when(mockAuthenticationContext.getProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN)).thenReturn(accessToken);
+        when(mockAuthenticationContext.getProperty(OIDCAuthenticatorConstants.ID_TOKEN)).thenReturn(idToken);
+
+        setParametersForOAuthClientResponse(mockOAuthClientResponse, accessToken, idToken);
+
+        mockStatic(OpenIDConnectAuthenticatorServiceComponent.class);
+        when(OpenIDConnectAuthenticatorServiceComponent.getRealmService()).thenReturn(mockRealmService);
+        when(mockRealmService.getTenantManager()).thenReturn(mockTenantManger);
+        when(mockTenantManger.getTenantId(anyString())).thenReturn(1234);
+
+        when(mockRealmService.getTenantUserRealm(-1234)).thenReturn(mockUserRealm);
```

Comment:
Better to use `getTenantUserRealm(anyInt())`

Reply:
This gives a null pointer